alacritty/src/config/bindings.rs

```diff
@@ -319,6 +397,34 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
         F19; Action::Esc("\x1b[33~".into());
         F20; Action::Esc("\x1b[34~".into());
         NumpadEnter; Action::Esc("\n".into());
+        Escape, ModifiersState::CTRL;                              Action::ToggleKeyboardMode;
+        Escape,                        +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
```

Comment:
@kchibisov I get where you're coming from, but what binding are you imagining for leaving this mode then? I'm not sure we _need_ a way to cancel an action or anything like that, clearing the selection can be done by toggling the active selection type (though if you don't know which selection mode you're in this can be non-trival).

Reply:
You already have **3** bindings to leave this mode. I feel like removing one of them could make sense, right? I'm pressing `i` tbh, but I accidentally clock on `ESC`, because I got used to it in vim to reset selection.

Reply:
I'm conflicted on this... On one hand we want to make it easy for a user to get out of this mode, since it'd be **very** frustrating to get stuck here. However, we also want to be somewhat minimal with out defaults.

I like following the vim bindings, so `i` makes sense to get back into what I've been calling "normal mode". Escape is also good, because it is used for mode changes in vim as well (even if they are slightly different, I don't see this as an issue in this case). The only binding I would remove is `q` actually, though I don't have strong feelings about that.

The fact that you leave selection mode in vim with ESC instead of `Shift + v` seems more like a you issue, not an issue with this design. In fact, I'd ague it's correct. ESC gets you back to the default mode, which is in our case "normal mode", which is more like vim's insert mode. If you want to maintain your mode and remove the selection, use the action to toggle it directly.

Reply:
After investigating what other terminals like URxvt and Termite do and based on my own intuition, I think I'm going to go with @nixpulvis here and make the decision to keep Escape to leave the mode completely. Though of course the beauty of this implementation is that you can freely remap this yourself.

I've also removed the `q` binding because that really was just mainly me throwing way too many bindings at it.

Reply:
While I do understand the idea behind your solution. It doesn't solve my particular problem of following vim behavior for vi like bindings. So if you decide to keep behavior of `ESC`, ok, but, please, add `ClearSelection` action, so I can actually have something vim-like and intuitive with my bindings..

Reply:
I thought there already was some, makes sense though.